adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/BasicActionMapper.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.justice.services.adapter.rest;
+
+import static java.lang.String.format;
+import static java.util.Collections.emptyMap;
+import static javax.ws.rs.HttpMethod.GET;
+
+import uk.gov.justice.services.adapter.rest.exception.BadRequestException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+
+
+public class BasicActionMapper {
+
+    private static final String MEDIA_TYPE_PREFIX = "application/vnd.";
+
+    private Map<String, Map<String, String>> methodToMediatypeAndActionMap = new HashMap<>();
+
+    protected void add(final String methodName, final String mediaType, String actionName) {
+        Map<String, String> mediaTypeToActionNameMap = methodToMediatypeAndActionMap.get(methodName);
+        if (mediaTypeToActionNameMap == null) {
+            mediaTypeToActionNameMap = new HashMap<>();
+            methodToMediatypeAndActionMap.put(methodName, mediaTypeToActionNameMap);
+        }
+        mediaTypeToActionNameMap.put(mediaType, actionName);
+
+    }
+
+    public String actionOf(final String methodName, final String httpMethod, final HttpHeaders headers) {
+        final Map<String, String> mediaTypeToActionMap = methodToMediatypeAndActionMap.getOrDefault(methodName, emptyMap());
```

Comment:
Do we need to pull this into a map first and then perform another get? (we're not null checking etc) can we just do it in one go?